### ISSUE TITLE
sssom merge method does not handle cases other than n=2

### ISSUE THREAD

**Issue by matentzn** [mapping-commons/sssom-py#147]
The cli.py:merge method makes some assumptions about the numbers of allowed parameters.

The function should take any number of inputs, (including 1!) and still return the right results.

- [x] Add test for 1 and 3 inputs
- [x] Generalise client and io methods to support arbitrary number of inputs

```
def merge(inputs: Tuple[str, str], output: str, reconcile: bool = True):
    """
    Merging msdf2 into msdf1,
        if reconcile=True, then dedupe(remove redundant lower confidence mappings) and
            reconcile (if msdf contains a higher confidence _negative_ mapping,
            then remove lower confidence positive one. If confidence is the same,
            prefer HumanCurated. If both HumanCurated, prefer negative mapping).

        Args:
            inputs: All MappingSetDataFrames that need to be merged
            output: SSSOM file containing the merged output
            reconcile (bool, optional): [description]. Defaults to True.

        Returns:

    """
    (input1, input2) = inputs[:2]
    msdf1 = read_sssom_table(input1)
    msdf2 = read_sssom_table(input2)
    merged_msdf = merge_msdf(msdf1, msdf2, reconcile)

    # If > 2 input files, iterate through each one
    # and merge them into the merged file above
    if len(inputs) > 2:
        for input_file in inputs[2:]:
            msdf1 = merged_msdf
            msdf2 = read_sssom_table(input_file)
            merged_msdf = merge_msdf(msdf1, msdf2, reconcile)

    # Export MappingSetDataFrame into a TSV
    write_table(merged_msdf, output)

```

**Comment by cthoyt** [comment]
Def a good place to use variadic arguments to make the interface more clean

```python
def merge(*inputs: str, output: str, reconcile: bool = True):
    msdfs = [read_sssom_table(i) for i in inputs]
    merged_msdf = merge_msdf(*msdfs, reconcile)
    write_table(merged_msdf, output)
```

**Comment by hrshdhgd** [comment]
Fixed by #195
